rbi/core/float.rbi

```diff
@@ -624,8 +624,13 @@ class Float < Numeric
   sig {params(base: Integer).returns(String)}
   def to_s(base=10); end
 
-  sig {returns(Integer)}
-  def truncate(); end
+  sig do
```

Comment:
While we don't support overloading for user-code, we do support it for standard library.
It seems that a better signature could be
```
sig {returns(Integer)}
sig do 
  params(
      arg0: Integer
    )
    .returns(T.any(Integer, Float))
  end
```

This will indicate that give no arguments, it returs an Integer, otherwise, either an Integer or a Float.
WDYT?

Reply:
Looks good.

What's the best way to test such changes?

Reply:
We used to test majority of them inside stdlib, but it's no longer sustainable to have a test-per-every-method in standard library.

I guess there should be some line where tests stop.
If there's some specific pattern that you care about, please add it as test_corpus/exp test: https://github.com/sorbet/sorbet#test_corpus-tests

but for this I'd be fine without a test so far.